File: server-mh/src/Match/Handler/Models/MatchState.ts
import { Phase, PlayerColor } from "../Enums";
import { Board } from "./Board";
import { Player } from "./Player";
import { TurnState } from "./TurnState";
import { DiceState } from "./DiceState";
import { MatchLabel } from "../MatchLabel";
import { GameActionData } from "../Actions/Datas";

export class MatchState {
    public tickCounter:number=0;  
    public board: Board;
    public players: Player[];
    public winnerList: PlayerColor[];
    public turnState: TurnState;
    public diceState: DiceState;
    public availableActions: GameActionData[]|undefined;
    public selectedAction:number=0;
    public currentPhase: Phase|null;
    public pendingPhase: Phase | null
    public matchEnd: boolean;
    public matchFinish:boolean;
    public label:MatchLabel;
    public version: number;

    constructor(
        board: Board,
        turnState: TurnState,
        diceState: DiceState,
        players: Player[] = [],
        label: MatchLabel = new MatchLabel()
    ) {
       
        this.board = board;
        this.players = players;
        this.winnerList = [];
        this.turnState = turnState;
        this.diceState = diceState;
        this.currentPhase = null;
        this.pendingPhase = Phase.Start;
        this.matchEnd=false,
        this.matchFinish=false;
        this.label=label;
        this.version = 1;
    }
}

File: Client-unity/Ludo/Assets/Scripts/OfflineMatchHandler/MatchState.cs
using System.Collections.Generic;

namespace Ludo.Offline
{
    public class MatchState
    {
        public int TickCounter;

        public Board Board;

        public Player[] Players;

        public PlayerColor[] WinnerList;

        public TurnState TurnState;

        public DiceState DiceState;

        public List<GameAction> AvailableActions;

        public int SelectedAction;

        public Phase? CurrentPhase;

        public Phase? PendingPhase;

        public bool MatchEnd;

        public bool MatchFinish;

        public MatchLabel Label;

        public int Version;

       public MatchState(
            Board board,
            TurnState turnState,
            DiceState diceState,
            Player[] players,
            MatchLabel label
            )
        {
            Board = board;
            TurnState = turnState;
            DiceState = diceState;
            Players = players;
            Label = label;
            CurrentPhase = Phase.Start;
            PendingPhase = null;

            WinnerList = new PlayerColor[0];
        }
    }
}